golint warnings shouldn't be ignored

`make verify` currently runs `golint` as a part of `gometalinter` which is configured to mask warnings.
This has created a lot of code which doesn't follow the suggested Go coding style. This needs to be fixed.

@kshlm For some reason, make verify gets stuck on my system for a very long time. Running `gometalinter` directly doesn't run most of the other checks.

```
[ppai@gd2-1 glusterd2]$ gometalinter 
main.go:19::warning: cyclomatic complexity 11 of function main() is high (> 10) (gocyclo)
config.go:15:2:warning: const defaultRpcAddress should be defaultRPCAddress (golint)
config.go:16:2:warning: const defaultRpcPort should be defaultRPCPort (golint)
main.go:64:5:warning: should omit comparison to bool constant, can be simplified to !gdctx.Restart (gosimple)
config.go:46::warning: Errors unhandled.,LOW,HIGH (gas)
WARNING: deadline exceeded by linter aligncheck on . (try increasing --deadline)
WARNING: deadline exceeded by linter varcheck on . (try increasing --deadline)
WARNING: deadline exceeded by linter interfacer on . (try increasing --deadline)
WARNING: deadline exceeded by linter unconvert on . (try increasing --deadline)
WARNING: deadline exceeded by linter structcheck on . (try increasing --deadline)
WARNING: deadline exceeded by linter errcheck on . (try increasing --deadline)
```

--- a/Makefile
+++ b/Makefile
@@ -37,6 +37,7 @@ vendor-update:
 	@echo
 
 verify: check-reqs
+	@./scripts/lint-check.sh
 	@gometalinter -D gotype -E gofmt --errors --deadline=5m -j 4 $$(glide nv)
 
 test:

--- a/etcdmgmt/server.go
+++ b/etcdmgmt/server.go
@@ -23,7 +23,7 @@ func StartEmbeddedEtcd(cfg *embed.Config) error {
 	defer etcdInstance.Unlock()
 
 	if etcdInstance.etcd != nil {
-		return errors.New("An instance of etcd embedded server is already running.")
+		return errors.New("An instance of etcd embedded server is already running")
 	}
 
 	// Start embedded etcd server
@@ -43,7 +43,7 @@ func StartEmbeddedEtcd(cfg *embed.Config) error {
 		return nil
 	case <-time.After(42 * time.Second):
 		etcd.Server.Stop() // trigger a shutdown
-		return errors.New("Etcd embedded server took too long to start!")
+		return errors.New("Etcd embedded server took too long to start")
 	case err := <-etcd.Err():
 		return err
 	}
@@ -68,12 +68,12 @@ func DestroyEmbeddedEtcd() error {
 
 	err := os.RemoveAll(etcdConfig.Dir)
 	if err != nil {
-		return errors.New("Could not delete etcd data dir.")
+		return errors.New("Could not delete etcd data dir")
 	}
 
 	err = os.RemoveAll(etcdConfig.WalDir)
 	if err != nil {
-		return errors.New("Could not delete etcd WAL dir.")
+		return errors.New("Could not delete etcd WAL dir")
 	}
 
 	os.Remove(EtcdConfigFile)

--- a/scripts/lint-check.sh
+++ b/scripts/lint-check.sh
@@ -1,0 +1,16 @@
+#/bin/bash
+
+# NOTE: This script needs be run from the root of the GD2 repository
+
+# Find all Go source files in the repository, that are not vendored or generated
+# and then run golint on them
+
+RETVAL=0
+
+for file in $(find . -path ./vendor -prune -o -type f -name '*.go' -not -name '*.pb.go' -print); do
+  golint -set_exit_status $file
+  if [ $? -eq 1 -a $RETVAL -eq 0 ]; then
+    RETVAL=1
+  fi
+done
+exit $RETVAL

--- a/transaction/context.go
+++ b/transaction/context.go
@@ -52,14 +52,6 @@ func NewCtxWithLogFields(fields log.Fields) *txnCtx {
 	return c
 }
 
-// NewTxnCtxWithPrefix returns a new context with the store prefix set
-func NewCtxWithPrefix(prefix string) *txnCtx {
-	c := NewCtx()
-	c.prefix = prefix
-
-	return c
-}
-
 // NewTxnCtx returns a new empty TxnCtx with no parent, no associated data and the default logger.
 func (c *txnCtx) NewCtx() *txnCtx {
 	return &txnCtx{

--- a/transaction/context.go
+++ b/transaction/context.go
@@ -27,8 +27,9 @@ type TxnCtx interface {
 	Prefix() string
 }
 
-type txnCtx struct {
-	parent *txnCtx
+// Tctx represents structure for transaction context
+type Tctx struct {
+	parent *Tctx
 	//data   map[string]interface{}
 	log       log.FieldLogger // Functions which are given this context must use this logger to log their data.
 	logFields log.Fields
@@ -37,14 +38,14 @@ type txnCtx struct {
 }
 
 // NewCtx returns a new empty TxnCtx with no parent, no associated data and the default logger.
-func NewCtx() *txnCtx {
-	return &txnCtx{
+func NewCtx() *Tctx {
+	return &Tctx{
 		log: log.StandardLogger(),
 	}
 }
 
 // NewCtxWithLogFields returns a new context with the logger set to log given fields
-func NewCtxWithLogFields(fields log.Fields) *txnCtx {
+func NewCtxWithLogFields(fields log.Fields) *Tctx {
 	c := NewCtx()
 	c.log = c.log.WithFields(fields)
 	c.logFields = fields
@@ -52,9 +53,9 @@ func NewCtxWithLogFields(fields log.Fields) *txnCtx {
 	return c
 }
 
-// NewTxnCtx returns a new empty TxnCtx with no parent, no associated data and the default logger.
-func (c *txnCtx) NewCtx() *txnCtx {
-	return &txnCtx{
+// NewCtx returns a new empty TxnCtx with no parent, no associated data and the default logger.
+func (c *Tctx) NewCtx() *Tctx {
+	return &Tctx{
 		parent:    c,
 		log:       c.log,
 		logFields: c.logFields,
@@ -63,7 +64,7 @@ func (c *txnCtx) NewCtx() *txnCtx {
 }
 
 // WithLogFields returns a new context with the logger set to log given fields
-func (c *txnCtx) WithLogFields(fields log.Fields) *txnCtx {
+func (c *Tctx) WithLogFields(fields log.Fields) *Tctx {
 	n := c.NewCtx()
 	n.log = n.log.WithFields(fields)
 	n.logFields = fields
@@ -72,7 +73,7 @@ func (c *txnCtx) WithLogFields(fields log.Fields) *txnCtx {
 }
 
 // WithPrefix returns a new context with the store prefix set
-func (c *txnCtx) WithPrefix(prefix string) *txnCtx {
+func (c *Tctx) WithPrefix(prefix string) *Tctx {
 	n := c.NewCtx()
 	n.prefix = prefix
 
@@ -81,7 +82,7 @@ func (c *txnCtx) WithPrefix(prefix string) *txnCtx {
 
 // Set attaches the given key-value pair to the context.
 // If the key exists, the value will be updated.
-func (c *txnCtx) Set(key string, value interface{}) error {
+func (c *Tctx) Set(key string, value interface{}) error {
 	json, e := json.Marshal(value)
 	if e != nil {
 		c.log.WithFields(log.Fields{
@@ -105,14 +106,14 @@ func (c *txnCtx) Set(key string, value interface{}) error {
 // SetNodeResult is similar to Set but prefixes the key with the node UUID
 // specified. This function can be used by nodes to store results of
 // transaction steps.
-func (c *txnCtx) SetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
+func (c *Tctx) SetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
 	storeKey := nodeID.String() + "/" + key
 	return c.Set(storeKey, value)
 }
 
 // Get gets the value for the given key if available.
 // Returns error if not found.
-func (c *txnCtx) Get(key string, value interface{}) error {
+func (c *Tctx) Get(key string, value interface{}) error {
 	storeKey := c.prefix + "/" + key
 	b, e := gdctx.Store.Get(storeKey)
 	if e != nil {
@@ -135,13 +136,13 @@ func (c *txnCtx) Get(key string, value interface{}) error {
 // GetNodeResult is similar to Get but prefixes the key with node UUID
 // specified. This function can be used by the transaction initiator node to
 // fetch results of transaction step run on remote nodes.
-func (c *txnCtx) GetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
+func (c *Tctx) GetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
 	storeKey := nodeID.String() + "/" + key
 	return c.Get(storeKey, value)
 }
 
 // Delete deletes the key and attached value
-func (c *txnCtx) Delete(key string) error {
+func (c *Tctx) Delete(key string) error {
 	storeKey := c.prefix + "/" + key
 	e := gdctx.Store.Delete(storeKey)
 	if e != nil {
@@ -155,12 +156,12 @@ func (c *txnCtx) Delete(key string) error {
 }
 
 // Logger returns the Logrus logger associated with the context
-func (c *txnCtx) Logger() log.FieldLogger {
+func (c *Tctx) Logger() log.FieldLogger {
 	return c.log
 }
 
 // Prefix returns the prefix to be used for storing values
-func (c *txnCtx) Prefix() string {
+func (c *Tctx) Prefix() string {
 	return c.prefix
 }
 
@@ -172,13 +173,13 @@ func (c *txnCtx) Prefix() string {
 // but JSON is simpler
 
 type expContext struct {
-	Parent    *txnCtx
+	Parent    *Tctx
 	LogFields log.Fields
 	Prefix    string
 }
 
 // MarshalJSON implements the json.Marshaler interface
-func (c *txnCtx) MarshalJSON() ([]byte, error) {
+func (c *Tctx) MarshalJSON() ([]byte, error) {
 	ac := expContext{
 		Parent:    c.parent,
 		LogFields: c.logFields,
@@ -189,7 +190,7 @@ func (c *txnCtx) MarshalJSON() ([]byte, error) {
 }
 
 // UnmarshalJSON implements the json.Unmarshaler interface
-func (c *txnCtx) UnmarshalJSON(d []byte) error {
+func (c *Tctx) UnmarshalJSON(d []byte) error {
 	var ac expContext
 
 	e := json.Unmarshal(d, &ac)

--- a/transaction/context_mock.go
+++ b/transaction/context_mock.go
@@ -6,31 +6,32 @@ import (
 	"github.com/pborman/uuid"
 )
 
-// MockCtx implements a dummy context type that can be used in tests
-type mockTxnCtx struct {
+// MockTctx implements a dummy context type that can be used in tests
+type MockTctx struct {
 	data map[string]interface{}
 }
 
-func NewMockCtx() *mockTxnCtx {
-	return &mockTxnCtx{
+// NewMockCtx returns a new instance of MockTctx
+func NewMockCtx() *MockTctx {
+	return &MockTctx{
 		data: make(map[string]interface{}),
 	}
 }
 
 // Set attaches the given key with value to the context. It updates value if key exists already.
-func (m *mockTxnCtx) Set(key string, value interface{}) error {
+func (m *MockTctx) Set(key string, value interface{}) error {
 	m.data[key] = value
 	return nil
 }
 
 // SetNodeResult is similar to Set but prefixes the key with the node UUID specified.
-func (c *mockTxnCtx) SetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
+func (m *MockTctx) SetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
 	storeKey := nodeID.String() + "/" + key
-	return c.Set(storeKey, value)
+	return m.Set(storeKey, value)
 }
 
 // Get gets the value for the given key. Returns an error if the key is not present
-func (m *mockTxnCtx) Get(key string, value interface{}) error {
+func (m *MockTctx) Get(key string, value interface{}) error {
 	v, ok := m.data[key]
 	if !ok {
 		return errors.New("key not present")
@@ -40,23 +41,23 @@ func (m *mockTxnCtx) Get(key string, value interface{}) error {
 }
 
 // GetNodeResult is similar to Get but prefixes the key with node UUID specified.
-func (m *mockTxnCtx) GetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
+func (m *MockTctx) GetNodeResult(nodeID uuid.UUID, key string, value interface{}) error {
 	storeKey := nodeID.String() + "/" + key
 	return m.Get(storeKey, value)
 }
 
 // Delete deletes the key and value
-func (m *mockTxnCtx) Delete(key string) error {
+func (m *MockTctx) Delete(key string) error {
 	delete(m.data, key)
 	return nil
 }
 
 // Logger returns a dummy logger
-func (m *mockTxnCtx) Logger() log.FieldLogger {
+func (m *MockTctx) Logger() log.FieldLogger {
 	return log.New()
 }
 
 // Prefix returns the prefix to be used for storing values
-func (m mockTxnCtx) Prefix() string {
+func (m MockTctx) Prefix() string {
 	return "mock"
 }

--- a/transaction/rpc-client.go
+++ b/transaction/rpc-client.go
@@ -78,7 +78,7 @@ func RunStepOn(step string, node uuid.UUID, c TxnCtx) (TxnCtx, error) {
 		return nil, errors.New(rsp.Error)
 	}
 
-	rspCtx := new(txnCtx)
+	rspCtx := new(Tctx)
 	err = json.Unmarshal(rsp.Resp, rspCtx)
 	if err != nil {
 		logger.WithError(err).Error("failed to JSON unmarhsal transaction context")

--- a/transaction/rpc-service.go
+++ b/transaction/rpc-service.go
@@ -19,7 +19,7 @@ func init() {
 
 // RunStep handles the incoming request. It executes the requested step and returns the results
 func (p *txnSvc) RunStep(rpcCtx context.Context, req *TxnStepReq) (*TxnStepResp, error) {
-	var ctx txnCtx
+	var ctx Tctx
 
 	err := json.Unmarshal(req.Context, &ctx)
 	if err != nil {

